### DISCUSSION TITLE
High Availability Concentrators - Proxmox Cluster - Homeassistant

### DISCUSSION THREAD

**Discussion by ckglobalroaming** [Koenkk/zigbee2mqtt#18332] [Devices]
Has anyone had any luck with HA (High Availability) for Concentrators?

At home I have a ProxMox Cluster with 5 x Host Nodes. These nodes are running a Ceph HA file store. I am wanting to run Homeassistant in HA mode so the server will auto migrate to one of the five Nodes for redundancy,

The challenge here is the Sonoff Concentrator is physically plugged into one of the Server Nodes and mapped in Proxmox to the Homeassistant VM. If we enable High Availability and the VM migrates to another Server Node then the Sonnoff will not be there.

### **Question:**
**Q:/** Is there any way to plug another concentrator into another server node (perfect world each server x4) so that every server has a concentrator and is available...

**Q:/** can we manage the USB sonnofs via linux scripting using the linux uhubctl command?
**Q:/** should the Sonoff be controlled externally

### **Possible Solution:**
I can install a Sonoff Concentrator on each Proxmox Cluster Node. Each of them can have ther power externally controlled. I can map them to the Home Assistant VM based on the vendorID as a wild card.

### **Curve Ball Issues:**
If I enable the power to the Sonoff when the VM is booted, 

- Will the VM see the ttyUSB0
- Will it always mount as ttyUSB0
- Will Zigbee2MQTT see it and use it?
- What implications are there for zigbee network rediscovery on different concentrators?

Any input is appreciated, has anyone tried to tackle this? Is ther a better way to approach this? Should it be an external pair of concentrators based on raspi's or network concentrators???

Thanks for taking the time to ready this.

thanks Craig.

**Comment by Matriciel** [comment]
You install a raspberry with Sonoff USB dongle.
You install the packet USBIP witch permit to share USB peripheral via TCP/IP on Raspberry and on your Z2M VM.
This USB dongle is sharing witch your Z2M VM regardless of the Proxmox server used.
